library/firewall_lib.py

```diff
@@ -423,10 +525,66 @@ def main():
             )
         ):
             module.fail_json(
-                msg="The states present and absent can only be used in zone level operations (i.e. when no other parameters but zone and state are set)."
+                msg="present and absent only usable for zone or service operations"
             )
+
+        # Zone and service are incompatible when state is set to present or absent
+        if zone and service:
+            module.fail_json(msg="both zone and service while state present/absent")
+
+        # While short and description are options for new zones, they are unimplemented
         if target is None and zone is not None:
-            zone_operation = True
+            if any(
+                (
+                    service,
+                    description,
+                    short,
+                    port,
+                    source_port,
+                    helper_module,
+                    protocol,
+                    destination_ipv4,
+                    destination_ipv6,
+                )
+            ):
+                module.fail_json(
+                    msg="short, description, port, source_port, helper_module, "
```

Comment:
Does this work?  I thought you had to put this in parentheses like
```python
  msg=(
    "short, description, ..."
    "protocol, or ......"
    "and state ..."
  )
```
?

Reply:
It seems to work somwhow